### PR TITLE
ec2 _find_image_serial: support serials in YYYYMMDD.n format (SC-1182)

### PR DESCRIPTION
**Pull request by paride**

### CODE REVIEW

**Comment by paride** [comment]
I was pretty sure I explicitly tested support for `.n` serials, but apparently I didn't.

**Comment by paride** [comment]
CI failure looks unrelated:
```
$ sudo snap install lxd

error: cannot perform the following tasks:

- Start snap "lxd" (23270) services (systemctl command [start snap.lxd.activate.service] failed with exit status 1: Job for snap.lxd.activate.service failed because the control process exited with error code.

See "systemctl status snap.lxd.activate.service" and "journalctl -xe" for details.

)

The command "sudo snap install lxd" exited with 1.
```

**Comment by paride** [comment]
@lucasmoura at the moment `ami-08fb9772db27d3d4a` on `us-east-2` has serial `20220712.1`, if you want to test this live.

File: pycloudlib/ec2/cloud.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def _find_image_serial(
             raise Exception("Could not find image: {}".format(image_id))
 
         image_name = images["Images"][0].get("Name", "")
-        serial_regex = r"ubuntu/.*/.*/.*-(?P<serial>\d+)$"
```
Comment (lucasmoura) [inline pycloudlib/ec2/cloud.py:219]: I think we have an extra parenthesis here. After removing it, I can confirm this indeed fix the image serial for the provided image-id

Comment (paride) [reply pycloudlib/ec2/cloud.py:219]: Fixed, thanks.

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**